exonum-java-binding/core/rust/integration_tests/tests/service_factory_adapter.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate exonum_testkit;
+extern crate integration_tests;
+extern crate java_bindings;
+
+use java_bindings::exonum::helpers::fabric::ServiceFactory;
+use java_bindings::JavaServiceFactoryAdapter;
+
+#[test]
+fn test_callbacks() {
+    let mut factory_adapter =
+        JavaServiceFactoryAdapter::new("service_name".to_owned(), "artifact_path".to_owned());
+    assert_eq!(factory_adapter.service_name(), "service_name");
+
+    // Make sure it returns None for all commands except `Run`.
+    assert!(factory_adapter.command("generate-template").is_none());
+    assert!(factory_adapter.command("generate-config").is_none());
+    assert!(factory_adapter.command("generate-testnet").is_none());
+    assert!(factory_adapter.command("finalize").is_none());
+    assert!(factory_adapter.command("run-dev").is_none());
+    assert!(factory_adapter.command("generate-template").is_none());
+
+    // Make sure it returns Some for the `Run` command for the first time and `None` for any other call.
```

Comment:
I suppose among several instances of such factories _first_ must *always* return Some,
other — None: https://docs.rs/exonum/0.11.0/exonum/helpers/fabric/trait.ServiceFactory.html#method.command
The trait spec does not say "I will only call this method once".